crates/runtime/src/lib.rs

```diff
@@ -1046,6 +1046,15 @@ impl Runtime {
 
                 self.remove_dataset(&ds).await;
 
+                // Initialize file mode accelerator when reloading with file mode acceleration
+                // Fail when the length of successfully initiated dataset is not 1
+                if ds.is_file_accelerated() {
+                    let datasets = self.initialize_accelerators(&[Arc::clone(&ds)]).await;
+                    if datasets.len() != 1 {
```

Comment:
```suggestion
                    if datasets.is_empty() {
```

**nit:** The vec can only ever be 1 or 0 in this case. `.is_empty()` is more succinct syntax.